cleanrl/ppo_dna_atari_envpool.py

```diff
@@ -0,0 +1,474 @@
+# DNA: Proximal Policy Optimization with a Dual Network Architecture
+# https://arxiv.org/abs/2206.10027
+
+import argparse
+import os
+import random
+import time
+from collections import deque
+from copy import deepcopy
+from distutils.util import strtobool
+
+import envpool
+import gym
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.distributions import Categorical, kl_divergence
+from torch.utils.tensorboard import SummaryWriter
+
```

Comment:
You'll probably need to add 

> torch.backends.cuda.matmul.allow_tf32 = False

before the first torch operation, otherwise, you may have issues with 30xx series cards. This issue has been fixed in torch 1.12, but I see we are pinned to an earlier version. Without it, some operations will be reduced to 10-bits of precision (around 3 sig fig) which may cause problems with, for example, the observation normalization. See (https://pytorch.org/docs/stable/notes/cuda.html#tf32-on-ampere) for a discussion on this issue.